driver-examples/osgi/src/test/java/com/datastax/driver/osgi/VersionProvider.java

```diff
@@ -25,11 +28,23 @@
  */
 public class VersionProvider {
 
+    private static final Pattern versionPattern = Pattern.compile(("(\\d+.\\d+\\.\\d+)(.*)"));
+
     private static String PROJECT_VERSION;
     static {
         String bundleName = Manifests.read("Bundle-SymbolicName");
         if (bundleName.equals("com.datastax.driver.osgi")) {
-            PROJECT_VERSION = Manifests.read("Bundle-Version").replaceAll("\\.SNAPSHOT", "-SNAPSHOT");
+            String bundleVersion = Manifests.read("Bundle-Version");
+            Matcher matcher = versionPattern.matcher(bundleVersion);
+            if(matcher.matches() && matcher.groupCount() == 2) {
+                String majorVersion = matcher.group(1);
```

Comment:
It seems that we always enter the if block, even for simple versions such as `1.2.3` because in these cases the capturing group `(.*)` captures an empty string.